AppCenter/AppCenter/Internals/Util/MSJwtClaims.m

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import "MSAppCenterInternal.h"
+#import "MSJwtClaims.h"
+#import "MSLogger.h"
+
+static NSString *const kMSJwtPartsSeparator = @".";
+static NSString *const kMSSubjectClaim = @"sub";
+static NSString *const kMSExpirationClaim = @"exp";
+
+@implementation MSJwtClaims
+
+- (instancetype)initWithSubject:(NSString *)subject expirationDate:(NSDate *)expirationDate {
+  self = [super init];
+  if (self) {
+    _subject = subject;
+    _expirationDate = expirationDate;
+  }
+  return self;
+}
+
++ (MSJwtClaims *)parse:(NSString *)jwt {
+  NSArray *parts = [jwt componentsSeparatedByString:kMSJwtPartsSeparator];
+  if ([parts count] < 2) {
```

Comment:
Hmm, looks like you're right.  I thought that JWTs didn't always require all three sections but it looks like they do: https://tools.ietf.org/id/draft-jones-json-web-token-01.html (Search for "When validating a JWT the following steps MUST be taken")

@guperrot we have the same bug in Android: https://github.com/microsoft/appcenter-sdk-android/pull/1240/files#diff-c86e6f6d9a168af1ce6ac45f63d50114R48

Reply:
Hmm, actually looking at this again, it seems that we only check for 2 parts because we don't care about what's in the last part because we don't do anything with it.  Is this OK?  @jaelim-ms

Reply:
It's not a full fledge jwt lib. We don't check headers and signature blocks client side, this will be done by ingestion. In SDK we just do the minimum to get sub and exp for auth token context.

Reply:
If Ingestion requires signature block, we can check it here and fail it earlier. It would things make harder to investigate any issues caused by this block. I would rather displaying error message in device console than looking up error messages in Ingestion if signature block is missing.

Reply:
Never mind. We need the log even though JWT isn't valid.